src/hotspot/share/opto/divnode.cpp

```diff
@@ -1084,12 +1145,98 @@ const Type* ModINode::Value(PhaseGVN* phase) const {
 
 //=============================================================================
 //------------------------------Idealize---------------------------------------
-Node *UModINode::Ideal(PhaseGVN *phase, bool can_reshape) {
+
+template <typename TypeClass, typename Unsigned>
+static Node* unsigned_mod_ideal(PhaseGVN* phase, bool can_reshape, Node* mod) {
   // Check for dead control input
-  if( in(0) && remove_dead_region(phase, can_reshape) )  return this;
+  if (mod->in(0) && mod->remove_dead_region(phase, can_reshape)) {
+    return mod;
+  }
+  // Don't bother trying to transform a dead node
+  if (mod->in(0) && mod->in(0)->is_top()) {
```

Comment:
```suggestion
  if (mod->in(0) != nullptr && mod->in(0)->is_top()) {
```

https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md
`Do not use ints or pointers as (implicit) booleans with &&, ||, if, while. Instead, compare explicitly, i.e. if (x != 0) or if (ptr != nullptr), etc.`

Reply:
There are more instances in the code. I know you copied it, but when we touch code we make sure to adhere to the new rules ;)